stac_model/utils.py

```diff
@@ -0,0 +1,215 @@
+    raise ValueError("Could not determine output channels from model weights.")
+
+
+def from_torch(
+    model: nn.Module,
+    *,
+    weights: Optional[object] = None,
+    item_id: str = "torch-model",
+    bbox: Optional[list[float]] = None,
+    geometry: Optional[dict] = None,
+    links: Optional[list[dict]] = None,
+    datetime_range: tuple[str, str] = (
+        "2015-06-23T00:00:00Z",  # Sentinel-2A launch date (first Sentinel-2 data available)
+        "2024-08-27T23:59:59Z",  # Dataset publication date Fields of The World (FTW)
+    ),
+) -> ItemMLModelExtension:
+    total_params = sum(p.numel() for p in model.parameters())
+    arch = f"{model.__class__.__module__}.{model.__class__.__name__}"
+    task = {TaskEnum.CLASSIFICATION}
+
+    # Extra metadata only found in weights of torchgeo models
+    has_meta = weights is not None and hasattr(weights, "meta")
+    state_dict = model.state_dict()
+
+    if has_meta:
+        in_chans = weights.meta.get("in_chans", get_input_channels(state_dict))
+        num_classes = weights.meta.get("num_classes", get_output_channels(state_dict))
+    else:
+        in_chans = get_input_channels(state_dict)
+        num_classes = get_output_channels(state_dict)
+
+    input_shape = [1, in_chans, 224, 224]
+    output_shape = [1, num_classes]
+
+    input_struct = InputStructure(
+        shape=input_shape,
+        dim_order=["batch", "channel", "height", "width"],
+        data_type="float32",
+    )
+
+    if has_meta and "bands" in weights.meta:
+        bands = weights.meta["bands"]
+    else:
+        bands = [f"band_{i}" for i in range(input_shape[1])]
+
+    model_input = ModelInput(
+        name="model_input",
+        bands=bands,
+        input=input_struct,
+        resize_type=None,
+        pre_processing_function=None,
+    )
+
+    classes = getattr(
+        model,
+        "classes",
+        [
+            MLMClassification(value=i, name=f"class_{i}", description=f"Auto-generated class {i}")
+            for i in range(output_shape[-1])
+        ],
+    )
+
+    model_output = ModelOutput(
+        name="model_output",
+        tasks=task,
+        result=ModelResult(
+            shape=output_shape,
+            dim_order=["batch", "classes"],
+            data_type="float32",
+        ),
+        classes=classes,
+        post_processing_function=None,
+    )
+
+    mlm_props = MLModelProperties(
+        name=item_id,
+        architecture=arch,
+        tasks=task,
+        input=[model_input],
+        output=[model_output],
+        total_parameters=total_params,
+        pretrained=True,
+        pretrained_source=None,
+    )
+
+    bbox = bbox or [-7.88, 37.13, 27.91, 58.21]
+    geometry = geometry or {
+        "type": "Polygon",
+        "coordinates": [
+            [
+                [-7.88, 37.13],
+                [-7.88, 58.21],
+                [27.91, 58.21],
+                [27.91, 37.13],
+                [-7.88, 37.13],
+            ]
+        ],
+    }
```

Comment:
These look like they are specific to the selected example. If they are not available in the model, they should be input parameters to the function instead.